src/test/java/seedu/address/logic/parser/meetup/AddCommandParserTest.java

```diff
@@ -1,5 +1,159 @@
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.Messages;
+import seedu.address.logic.commands.meetup.AddCommand;
+import seedu.address.model.buyer.Name;
+import seedu.address.model.meetup.From;
+import seedu.address.model.meetup.MeetUp;
+import seedu.address.model.meetup.To;
+import seedu.address.testutil.meetup.MeetUpBuilder;
+
 public class AddCommandParserTest {
-    // TO-DO
+
+    private seedu.address.logic.parser.meetup.AddCommandParser parser = new AddCommandParser();
+
+    @Test
+    public void parse_allFieldsPresent_success() {
+        MeetUp expectedMeetUp = new MeetUpBuilder(PITCH_MEETUP).build();
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + MEETUP_NAME_DESC_PITCH
+                + MEETUP_INFO_DESC_PITCH + MEETUP_FROM_DESC_PITCH
+                + MEETUP_TO_DESC_PITCH, new AddCommand(expectedMeetUp));
+    }
+
+    @Test
+    public void parse_repeatedNonTagValue_failure() {
+        String validExpectedMeetUpString = MEETUP_NAME_DESC_PITCH + MEETUP_INFO_DESC_PITCH
+                + MEETUP_FROM_DESC_PITCH + MEETUP_TO_DESC_PITCH;
+
+        // multiple names
+        assertParseFailure(parser, MEETUP_NAME_DESC_NETWORKING + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_NAME));
+
+        // multiple infos
+        assertParseFailure(parser, MEETUP_INFO_DESC_NETWORKING + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_INFO));
+
+        // multiple froms
+        assertParseFailure(parser, MEETUP_FROM_DESC_NETWORKING + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_FROM));
+
+        // multiple tos
+        assertParseFailure(parser, MEETUP_TO_DESC_NETWORKING + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TO));
+
+        // multiple fields repeated
+        assertParseFailure(parser,
+                validExpectedMeetUpString + MEETUP_NAME_DESC_PITCH + MEETUP_INFO_DESC_PITCH
+                        + MEETUP_FROM_DESC_PITCH + MEETUP_TO_DESC_PITCH + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_NAME, PREFIX_INFO, PREFIX_FROM, PREFIX_TO));
+
+        // invalid value followed by valid value
+
+        // invalid name
+        assertParseFailure(parser, INVALID_MEETUP_NAME_DESC + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_NAME));
+
+        // invalid from
+        assertParseFailure(parser, INVALID_MEETUP_FROM_DESC + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_FROM));
+
+        // invalid to
+        assertParseFailure(parser, INVALID_MEETUP_TO_DESC + validExpectedMeetUpString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_TO));
+
```

Comment:
For completeness, do also add a // invalid info
 assertParseFailure(parser, INVALID_MEETUP_INFO_DESC + validExpectedMeetUpString,
                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_INFO));